search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+    "by" : "dateissued",
+    "order" : "asc"
+  },
+  "page": {
+    	"size": 5,
+    	"totalElements": 14,
+    	"totalPages": 3,
+    	"number": 0
+  },
+  "_embedded" : {
+    "searchResults" : [
+      {
+        "hitHighlights": {
+          "dc.description.abstract" : "This is the <em>very cool</em> abstract of this item",
+          "dc.publisher" : "My <em>very cool</em> publisher",
+        },
+        "_links" : {
+          "dspaceObject" : {
+            "href": "/api/core/items/9f3288b2-f2ad-454f-9f4c-70325646dcee"
+          }
+        },
+        "_embedded" : {
+          "dspaceObject" : {
+            "uuid": "9f3288b2-f2ad-454f-9f4c-70325646dcee",
+            "name": "Test Webpage",
+            "handle": "10673/4"
+          }
+        }
+      },
+      {
+        "hitHighlights": { },
+        "_links" : {
+          "dspaceObject" : {
+            "href": "/api/core/items/ff7ec3a4-0aab-418b-94fc-d0e8189084db"
+          }
+        },
+        "_embedded" : {
+          "dspaceObject" : {
+            "uuid": "ff7ec3a4-0aab-418b-94fc-d0e8189084db",
+            "name": "Test Item with no hit highlights",
+            "handle": "10673/5"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+      "first": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=0&size=5"
+      },
+      "self": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=0&size=5"
+      },
+      "next": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=1&size=5"
+      },
+      "last": {
+        "href": "/api/discover/search/<:dso-type>/objects?query=my+query&scope=9076bd16-e69a-48d6-9e41-0238cb40d863&title.notcontains=abcd&author.authority=1234&page=2&size=5"
+      }
+  }
+}
+```
+
+### Matching facet search results
+**/api/discover/search/<:dso-type>/facets**
+
+This endpoint returns a list of facet values and counts based on the current query. The result can be refined using the following parameters:
+* `query`: The discovery search string to will be used to match records.
+* `scope`: UUID of a specific DSpace container (site, community or collection) to which the search has to be limited, e.g. `scope=9076bd16-e69a-48d6-9e41-0238cb40d863`.
+* `<:filter-name>.<:filter-operator>`: Advanced search filter that has to be used to filter the result set. The `filter-name` and `filter-operator` must match a value returned by parent search endpoint (see above). For example `author.authority=5df05073-3be7-410d-8166-e254369e4166` or `title.notcontains=rainbows`.
+* There is no paging here since the number of facet values displayed in the Discovery search screen is configured in the backend.
```

Comment:
You are right and as you can see this is done in the `/api/discover/facets/<:facet-name>` endpoint. This is just a "utility" endpoint to give you the values for all configured values at the same time (so that the UI doesn't have to make a call for each separate facet). 

Since the result contains values of multiple facts, I don't think it is possible to paginate (Do you want to page over the author facet? Or over the subject facet? Because both are in the response).